### DISCUSSION TITLE
Hashes different in pipeline vs local

### DISCUSSION THREAD

**Discussion by eosgortor** [nrwl/nx#13195] [General]
I saw hashes being different in pipelines vs local using the same commit and command. Now I wonder what makes this difference. 

The documenation is quite unspecific when it comes to what is actually hashed. When looking around I found theses items to be included in my case (npm, no workspaces):

- Project:
	- All files under git control in the project folder (not the ones in .gitignore):
		- file names
		- file content
	- Version of the project or projectName
	- Hashes of dependencies
- Root:
	- implicit dependencies
	- nx.json
	- package-lock.json
	- tsconfig.base.json
	- .gitignore
	- .nxignore
- Command
	- Project
	- Target
	- Configuration
	- Command (flags)
- Misc
	- Version of hasher

Is there something else? Maybe the operating system?

As hashing is quite intransparent at the moment, this is also a feature request. Could there be some tooling to see what is actually hashed for a task to debug this kind of problems? Maybe by verbose logs, a special command flag or an extra command.

**Comment by govindappaarun** [comment]
Were you able to find the answer for this question from any other sources?, 
we also face same issue for our repo where Jenkins CI hash are not matching with local

**Reply by eosgortor** [reply]
No, we are living with it for the moment.
